conans/paths/__init__.py

```diff
@@ -12,21 +12,31 @@
 
 
 def get_conan_user_home():
-    def _read_user_home_from_rc():
+
+    def _find_conanrc_file():
+        path = Path(os.getcwd())
+        while path.is_dir() and len(path.parts) > 1:  # finish at '/'
+            conanrc_file = path / ".conanrc"
+            if conanrc_file.is_file():
+                return conanrc_file
+            else:
+                path = path.parent
+
+    def _user_home_from_conanrc_file():
         try:
-            conanrc_path = os.path.join(os.getcwd(), ".conanrc")
+            conanrc_path = _find_conanrc_file()
             values = {k: str(v) for k, v in
                       (line.split('=') for line in open(conanrc_path).read().splitlines() if
```

Comment:
I was wondering if this file descriptor keeps open until the process finish but apparently is doing well, it gets closed after the `for` loop. Just for my own knowledge, is that a feature of python3? it was always like that? I cannot find documentation about the automatic close. If it was accidental, should we read the contents first?

Reply:
I found some info on Stack Overflow and apparently the garbage collection will handle this in different ways depending on the Python implementation, but I think that if we want to make sure that this won't be problematic in the future we should probably change it. I think I'll better remove the comprehension.